x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java

```diff
@@ -1155,6 +1163,75 @@ private void assertSearchableSnapshotStats(String indexName, boolean cacheEnable
+        final CyclicBarrier cyclicBarrier = new CyclicBarrier(2);
+
+        final String indexName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+        createAndPopulateIndex(indexName, indexSettingsNoReplicas(1).put(INDEX_SOFT_DELETES_SETTING.getKey(), true));
+
+        final String repositoryName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+        createRepository(repositoryName, "fs");
+
+        final SnapshotId snapshotOne = createSnapshot(repositoryName, "snapshot-1", List.of(indexName)).snapshotId();
+        assertAcked(indicesAdmin().prepareDelete(indexName));
+
+        // block the cluster state update thread, so we have the opportunity to inspect the mount/restore tasks' metadata
+        ClusterService clusterService = internalCluster().getCurrentMasterNodeInstance(ClusterService.class);
+        clusterService.submitUnbatchedStateUpdateTask("block master service", new ClusterStateUpdateTask() {
+            @Override
+            public ClusterState execute(ClusterState currentState) {
+                safeAwait(cyclicBarrier);   // notify test that we're blocked
+                safeAwait(cyclicBarrier);   // wait to be unblocked by test
+                return currentState;
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                throw new AssertionError("block master service", e);
+            }
+        });
+
+        // wait for master thread to be blocked
+        safeAwait(cyclicBarrier);
+
+        try {
+            final MountSearchableSnapshotRequest mountRequest = new MountSearchableSnapshotRequest(
+                indexName,
+                repositoryName,
+                snapshotOne.getName(),
+                indexName,
+                Settings.EMPTY,
+                Strings.EMPTY_ARRAY,
+                false,
+                MountSearchableSnapshotRequest.Storage.FULL_COPY
+            );
+
+            client().execute(MountSearchableSnapshotAction.INSTANCE, mountRequest, ActionListener.noop());
+            assertBusy(() -> {
+                // only asks for tasks on the master node
+                var response = listTasks(internalCluster().getMasterName());
+                TaskInfo restoreSnapshotTask = response.getTasks()
+                    .stream()
+                    .filter(t -> t.action().equals(TransportRestoreSnapshotAction.TYPE.name()))
+                    .findAny()
+                    .orElseThrow(() -> new AssertionError("No restore snapshot task found"));
+                TaskInfo mountSnapshotTask = response.getTasks()
+                    .stream()
+                    .filter(t -> t.action().equals(MountSearchableSnapshotAction.NAME))
+                    .findAny()
+                    .orElseThrow(() -> new AssertionError("No mount snapshot task found"));
+                assertEquals(mountSnapshotTask.taskId(), restoreSnapshotTask.parentTaskId());
```

Comment:
There's no point retrying here if this fails, I'll move that out of the `assertBusy` somehow. Perhaps `assertBusy` isn't appropriate here, I just wanted to avoid polling the list tasks API forever.